src/Data/Map/Internal.purs

```diff
@@ -638,3 +640,17 @@ filterKeys predicate = filterWithKey $ const <<< predicate
 -- | on the value fails to hold.
 filter :: forall k v. Ord k => (v -> Boolean) -> Map k v -> Map k v
 filter predicate = filterWithKey $ const predicate
+
+-- | Apply a map of functions to a map of inputs.
+-- | If a key only exists on one side, then a default is used.
+applyWithDefault ::
+  forall k i o . Ord k =>
+  Map k (i -> o) -> Maybe (i -> o) ->
+  Map k i -> Maybe i ->
+  Map k o
+applyWithDefault fns defFn vals defVal =
+  let build k = Tuple k <$> ((lookup k fns <|> defFn) <*>
```

Comment:
Wouldn't it be possible to use `union` here, like the implementation in `Data.Align`? That way if we get a faster hedge union, this will improve for free too.

Reply:
Another thought: why not add this in the form of newtype `MapWithDefault k a = MapWithDefault a (Map k a)` with an `Applicative` instance? I think Conal has a Haskell library which does that.

Reply:
I'll have a think about the `union`, thanks.

Found the library you mentioned: https://hackage.haskell.org/package/total-map-0.0.6/docs/Data-TotalMap.html
I think both the type you propose and mine are useful for difference purposes.
With the one I proposed you get to model merges where one/both/neither have a default.